packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -32,6 +30,18 @@ type NotificationManager = import('./record-notification-manager').default;
 type DSModel = import('../ts-interfaces/ds-model').DSModel;
 type ShimModelClass = import('./model/shim-model-class').default;
 type DSModelClass = import('@ember-data/model').default;
+type RecordDataClass = typeof import('@ember-data/record-data/-private').RecordData;
+
+// we can't greedily use require as this causes
+// an order of operations issue which gives us undefined
```

Comment:
This comment seems to be describing a cycle, can you update the comment to make it clear where the actual cycle is (something like `foo/bar -> baz/qux -> foo/bar`)